### PR TITLE
Add wolfcrypt ML-DSA (dilithium) support.

### PR DESCRIPTION
**Pull request by philljj**
# Description

Adds wolfcrypt ML-DSA (wolfcrypt/src/dilithium.c) support to wolfBoot.

Also covers some missing testing for sim target with PQ.

# Config

Added sim ML-DSA example:
- config/examples/sim-ml-dsa.config

Use `ML_DSA_LEVEL=<num>` to configure the ML-DSA parameter set. See the config example and updated PQ docs for explanation of parameters.

# Docs

Updated `docs/PQ.md` to describe ML-DSA and all three NIST approved PQ sig algs we have implemented.

# Testing

- Added script `tools/scripts/sim-pq-sunnyday-update.sh`. This is to simplify sim-PQ sunnyday tests.
- Added LMS, XMSS, ML-DSA to `.github/workflows/test-sunnyday-simulator.yml`.
- Added renode-nrf52 ML-DSA-44 test to github workflow.

# Prereqs

Requires this PR to build:
- https://github.com/wolfSSL/wolfssl/pull/7983

Requires this PR to conform to FIPS 204:
- https://github.com/wolfSSL/wolfssl/pull/8016

Requires these PRs to update lib/wolfssl submodule to master:
- https://github.com/wolfSSL/wolfssl/pull/8053
- https://github.com/wolfSSL/wolfBoot/pull/509

### CODE REVIEW

File: src/image.c [hunk 1]
```diff
@@ -472,6 +472,98 @@ static void wolfBoot_verify_signature(uint8_t key_slot,
...
+        return;
+    }
+
+    /* Set the ML-DSA security level. */
+    ret = wc_MlDsaKey_SetParams(&ml_dsa, ML_DSA_LEVEL);
+
+    if (ret != 0) {
+        wolfBoot_printf("error: wc_MlDsaKey_SetParams(%d)" \
+                        " returned %d\n", ML_DSA_LEVEL, ret);
+        return;
```
Comment (dgarske) [inline src/image.c:509]: Prefer a single return flow to make sure `wc_MlDsaKey_Free(&ml_dsa);` gets called. Refactor function with `if (ret == 0)` logic and only have a single return at bottom.

File: tools/keytools/Makefile [hunk 2]
```diff
@@ -162,6 +162,18 @@ OBJS_REAL+=\
 	$(XMSSDIR)/utils.o
 endif
 
+# Add wolfcrypt ML-DSA (dilithium) implementation.
+ifeq ($(SIGN),ML_DSA)
+  OBJS_REAL+=$(WOLFDIR)/wolfcrypt/src/dilithium.o
+
+  CFLAGS += -D"WOLFSSL_DILITHIUM_FIPS204_DRAFT" \
```
Comment (dgarske) [inline tools/keytools/Makefile:169]: I would prefer these build options reside in tools/keytools/user_settings.h. Please move this and others if possible.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: tools/keytools/Makefile (hunk 2 above)
Comment (philljj) [reply tools/keytools/Makefile:169]: I'll move these, agree will be tidier

File: src/image.c [hunk 3]
```diff
@@ -472,6 +472,104 @@ static void wolfBoot_verify_signature(uint8_t key_slot,
...
+        else if (sig_len != IMAGE_SIGNATURE_SIZE) {
+            wolfBoot_printf("error: ML-DSA sig len mismatch: got %d bytes " \
+                   "expected %d\n", sig_len, IMAGE_SIGNATURE_SIZE);
+            ret = -1;
+        }
+    }
+
+    if (ret == 0) {
+        /* Now import pub key. */
+        ret = wc_MlDsaKey_ImportPubRaw(&ml_dsa, pubkey, pub_len);
```
Comment (dgarske) [inline src/image.c:542]:
```
davidgarske@Davids-MacBook-Pro-2 wolfBoot-alt % make
        [CC-sim] src/string.o
        [CC-sim] src/image.o
src/image.c:542:15: error: use of undeclared identifier 'out'
  542 |         ret = wc_MlDsaKey_ImportPubRaw(&ml_dsa, pubkey, pub_len);
      |               ^
lib/wolfssl/wolfssl/wolfcrypt/dilithium.h:753:32: note: expanded from macro 'wc_MlDsaKey_ImportPubRaw'
  753 |     wc_dilithium_import_public(out, outLen, key)
      |                                ^
src/image.c:542:15: error: use of undeclared identifier 'outLen'
lib/wolfssl/wolfssl/wolfcrypt/dilithium.h:753:37: note: expanded from macro 'wc_MlDsaKey_ImportPubRaw'
  753 |     wc_dilithium_import_public(out, outLen, key)
      |                                     ^
2 errors generated.
make: *** [src/image.o] Error 1
```

**Review by dgarske — DISMISSED** [review]

File: src/image.c (hunk 3 above)
Comment (dgarske) [reply src/image.c:542]:
```
cp config/examples/sim-ml-dsa.config .config
make distclean
make
```

Comment (philljj) [reply src/image.c:542]: Have to build with this branch:
- https://github.com/wolfSSL/wolfssl/pull/7983

**Review by danielinux — CHANGES_REQUESTED** [review]
Tested OK against wolfssl bf29b686

Please do:
- update wolfSSL submodule to master
- add simulator tests

**Comment by danielinux** [comment]
re: simulator tests

It looks like we don't have those tests for LMS/XMSS either. Can we perhaps extend `test-sunnyday-simulator.yml` workflow, adding cases for LMS, XMSS, ML-DSA?

**Comment by philljj** [comment]
> re: simulator tests
> 
> It looks like we don't have those tests for LMS/XMSS either. Can we perhaps extend `test-sunnyday-simulator.yml` workflow, adding cases for LMS, XMSS, ML-DSA?

I'll add them all. It will be good to have sim coverage for all PQ as well!

**Comment by philljj** [comment]
Can reproduce failure locally. Seems to be from introduction of wolf strdup over summer.

**Review by danielinux — APPROVED** [review]

**Pull request merged**